File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/houses/AccountHouseMessage.scala





















// Generated on 05/03/2013 10:34:43
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait AccountHouseMessage extends MessageDeserializer {
  val messageId = 6315

  protected trait Message extends BaseMessage with Serializable {
    val messageId = AccountHouseMessage.messageId

    val houses: Seq[AccountHouseInformations.TypeImpl]

    def serialize(buf: Buffer) {
      buf.writeUShort(houses.length)
      for (entry <- houses) {
        entry.serialize(buf)
      }
    }

  }

  protected def create(houses: Seq[AccountHouseInformations.TypeImpl]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val houses = {
      val builder = collection.immutable.Seq.newBuilder[AccountHouseInformations.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += AccountHouseInformations.deserialize(buf)
      }
      builder.result()
    }

    create(houses)
  }

}

object AccountHouseMessage extends AccountHouseMessage {
  case class MessageImpl(houses: Seq[AccountHouseInformations.TypeImpl]) extends Message
  type Target = MessageImpl

  override def create(houses: Seq[AccountHouseInformations.TypeImpl]) = MessageImpl(houses)

  def apply(houses: Seq[AccountHouseInformations.TypeImpl]) = create(houses)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/AccountHouseMessage.java


// Generated on 05/08/2013 19:37:46
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AccountHouseMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6315;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public AccountHouseInformations[] houses;
    
    public AccountHouseMessage() { }
    
    public AccountHouseMessage(AccountHouseInformations[] houses) {
        this.houses = houses;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(houses.length);
        for (AccountHouseInformations entry : houses) {
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        int limit = buf.readUShort();
        houses = new AccountHouseInformations[limit];
        for (int i = 0; i < limit; i++) {
            houses[i] = new AccountHouseInformations();
            houses[i].deserialize(buf);
        }
    }
    
}
